### ISSUE TITLE
Error signing up: "o.data is undefined"

### ISSUE THREAD

**Issue by elimisteve** [phrazzld/studymode#2]
Interesting project! There are _much_ shorter paths through the knowledge graph for learning all sorts of things, and we need more attempts to rapidly find and popularize them :ok_hand: .

Screenshot of error when trying to sign up:

![studymode-signup-error-Screenshot_2023-02-12_21-07-08](https://user-images.githubusercontent.com/139603/218375347-3cfa4484-1645-4482-89f2-9157b86b675f.png)

**Comment by elimisteve** [comment]
This is in Firefox btw. The JS console is empty.

**Comment by elimisteve** [comment]
I tried signing up again and got this error, so I guess my signup succeeded but there was an error parsing the response from the server or something?

![studymode-signup-error2-Screenshot_2023-02-12_21-14-25](https://user-images.githubusercontent.com/139603/218376343-21454906-84fc-4e5e-8548-d1c0525a1483.png)

**Comment by moomooskycow** [comment]
Thanks for taking a look at the project Steve! 

> There are much shorter paths through the knowledge graph for learning all sorts of things, and we need more attempts to rapidly find and popularize them 👌 .

Fully agree. The company I work for, Memre, does some pretty interesting stuff on the "optimize your personal review schedule" and "know what you know" side of things, and I think GPT and other generative AI projects enable effortless production of infinite learning content. The ultimate vision for all learning apps is pushbutton upload of any concept to your brain, Matrix style, and maybe this project can be a baby step in that direction.

It is super prototype alpha right now haha, so I appreciate you reporting the funk you run into. This issue is high priority -- sign up flow's gotta be smooth!

**Comment by moomooskycow** [comment]
I couldn't reproduce the `o.data` error, but I did get a `createMemreUserResponseJson.data is undefined` issue that I just fixed. Also added a redirect in the auth component so that if you're already logged in you get sent back to the home page -- should address that confusing second state you sent over (i.e. something fails but it's recoverable and you're still stuck on the auth page thinking you don't have an account).

Change is deploying now (this is my first Next.js project, Next.js and Vercel are pretty rad). I'll do some more testing on prod to verify that the signup flow works as expected before closing this issue.
